coins/bitcoin/src/rpc.rs

```diff
@@ -0,0 +1,538 @@
+        let args = handle_defaults(&mut ext_args, &defaults);
+        let hex: String = self.rpc_call("getblock", &args).await?;
+        let bytes: Vec<u8> = FromHex::from_hex(&hex)?;
+        Ok(bitcoin::consensus::encode::deserialize(&bytes)?)
+    }
+
+    pub async fn get_block_with_transactions(&self, block_hash: &bitcoin::BlockHash) -> anyhow::Result<GetBlockWithDetailResult> {
+        let mut ext_args = [into_json(block_hash)?, 2.into()];
+        let defaults = [null(), 1.into()];
+        let args = handle_defaults(&mut ext_args, &defaults);
+        let info: GetBlockWithDetailResult = self.rpc_call::<GetBlockWithDetailResult>("getblock", &args).await?;
+        Ok(info)
+    }
+
+    pub async fn get_best_block_hash(&self) -> Result<bitcoin::BlockHash> {
+        let info = self.rpc_call::<bitcoin::BlockHash>("getbestblockhash", &[]).await?;
+        Ok(info)
+    }
+
+    pub async fn get_spendable(
+        &self,
+        minconf: Option<usize>,
+        maxconf: Option<usize>,
+        addresses: Option<Vec<&str>>,
+        include_unsafe: Option<bool>,
+    ) -> anyhow::Result<Vec<ListUnspentResultEntry>> {
+        let mut ext_args = [
+            opt_into_json(minconf)?,
+            opt_into_json(maxconf)?,
+            opt_into_json(addresses)?,
+            opt_into_json(include_unsafe)?,
+        ];
+        let defaults = [into_json(1)?, into_json(9999999)?, null(), into_json(true)?];
+        let args = handle_defaults(&mut ext_args, &defaults);
+        let info = self.rpc_call("listunspent", &args).await?;
+        Ok(info)
+    }
+
+    pub async fn get_o_indexes(
+        &self,
+        minconf: Option<usize>,
+        maxconf: Option<usize>,
+        addresses: Option<Vec<&str>>,
+        include_unsafe: Option<bool>,
+    ) -> anyhow::Result<Vec<ListUnspentResultEntry>> {
+        let info: Vec<ListUnspentResultEntry> = self
+            .get_spendable(minconf, maxconf, addresses, include_unsafe)
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn lock_unspent(&self, outputs: &[OutPoint]) -> Result<bool> {
+        let outputs: Vec<_> = outputs
+            .iter()
+            .map(|o| serde_json::to_value(JsonOutPoint::from(*o)).unwrap())
+            .collect();
+        let info: bool = self
+            .rpc_call::<bool>("lockunspent", &[false.into(), outputs.into()])
+            .await?;
+        Ok(info)
+    }
+
+    pub async fn unlock_unspent(&self, outputs: &[OutPoint]) -> Result<bool> {
+        let outputs: Vec<_> = outputs
+            .iter()
+            .map(|o| serde_json::to_value(JsonOutPoint::from(*o)).unwrap())
+            .collect();
+        let mut ext_args = [true.into(), outputs.into()];
+        let args = handle_defaults(&mut ext_args, &[true.into(), empty_arr()]);
+        let info: bool = self.rpc_call::<bool>("lockunspent", &args).await?;
+        Ok(info)
+    }
+
+    pub async fn get_fee_per_byte(&self, conf_target: Option<usize>) -> anyhow::Result<u64> {
```

Comment:
Remove. While `estimatesmartfee` appears deterministic, that assumes a consistent best block which we can't guarantee, both due to competitors and due to latency.